engine/src/test/java/org/camunda/bpm/engine/test/api/mgmt/ManagementServiceGetTelemetryDataTest.java

```diff
@@ -94,16 +68,15 @@
   @Before
   public void setup() {
     configuration = engineRule.getProcessEngineConfiguration();
-    managementService = engineRule.getManagementService();
+    managementService = (ManagementServiceImpl) engineRule.getManagementService();
     runtimeService = engineRule.getRuntimeService();
     telemetryRegistry = configuration.getTelemetryRegistry();
     metricsRegistry = configuration.getMetricsRegistry();
 
-    clearMetrics();
-    configuration.getTelemetryRegistry().clear();
-
     defaultTelemetryData = new TelemetryDataImpl(configuration.getTelemetryData());
```

Comment:
If we clear the dynamic telemetry data (anything that is determined after engine start) after the test, there should be no leakage into other tests, or? You can considering creating a method in `ManagementServiceImpl` that does this, then we can reduce some more dependency on internals.

Reply:
I am not sure if I get your point. We should avoid leaking dynamic telemetry data into other tests or from previous tests into this one. This is why we clear the data before and after each test. Technically it would be enough to clear it in a `@BeforeClass` method and then after each test.
Do you suggest doing it differently?

The suggestion to add a method to `ManagementServiceImpl` makes sense.

Reply:
I wonder why we have the code that stores the engine-internal `TelemetryDataImpl` in a field before the test and restores it afterwards, since this makes a heavy assumption about implementation behavior. If we had a method `ManagementServiceImpl#createDynamicTelemetryData` or similar, that we can call before/after a test, then this would be no longer necessary.

Reply:
It's still not clear to me what your goal is.
The reason we store/restore the telemetry data is to ensure we don't modify it with the test.
Do you suggest not doing it or is your suggestion to do it through an API method in `ManagementServiceImpl`?

Reply:
As discussed, I will leave this part as is. We agreed that storing and restoring states this way relies on internals and makes assumptions about implementations. However, we use this pattern for other tests involving telemetry data. So ideally we should fix it everywhere. For now, let's keep it as it is.